nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -428,14 +436,49 @@ public static void checkTypeParameterNullnessForAssignability(
       AssignmentTree assignmentTree = (AssignmentTree) tree;
       rhsTree = assignmentTree.getExpression();
     }
+
+    if (rhsTree instanceof MethodInvocationTree) {
+      MethodInvocationTree methodInvocationTree = (MethodInvocationTree) rhsTree;
+      Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocationTree);
+      // update inferredTypes cache for assignments
+      // generic method call with no explicit generic arguments
+      if (methodSymbol.type instanceof Type.ForAll
+          && methodInvocationTree.getTypeArguments().isEmpty()) {
+        Type returnType = methodSymbol.getReturnType();
+        Map<TypeVariable, Type> genericNullness =
+            returnType.accept(new InferTypeVisitor(config), lhsType);
+        if (genericNullness != null) {
+          inferredTypes.put(methodInvocationTree, genericNullness);
+        }
+      }
+    }
     // rhsTree can be null for a VariableTree.  Also, we don't need to do a check
     // if rhsTree is the null literal
     if (rhsTree == null || rhsTree.getKind().equals(Tree.Kind.NULL_LITERAL)) {
       return;
     }
     Type rhsType = getTreeType(rhsTree, config);
 
-    if (lhsType != null && rhsType != null) {
+    if (rhsTree instanceof MethodInvocationTree) {
```

Comment:
Couldn't we move the code under the `if` at line 440 to also be under this `if`?  We'd still need to store `genericNullness` inside `inferredTypes` but then we could just use it immediately here to reconstruct the inferred type.